modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java

```diff
@@ -105,12 +106,10 @@ public void renderHTML(
 		RequestDispatcher requestDispatcher =
 			_servletContext.getRequestDispatcher("/select_folder.jsp");
 
-		long folderId = BeanParamUtil.getLong(
-			itemSelectorCriterion, (HttpServletRequest)servletRequest,
-			"folderId");
-		long selectedRepositoryId = BeanParamUtil.getLong(
-			itemSelectorCriterion, (HttpServletRequest)servletRequest,
-			"selectedRepositoryId");
+		long repositoryId = ParamUtil.getLong(
+			(HttpServletRequest)servletRequest, "repositoryId");
+		long folderId = ParamUtil.getLong(
+			(HttpServletRequest)servletRequest, "folderId");
```

Comment:
Can we change the impl of the get*URL methods in the display context so that all parameters go through the itemSelectorCriterion? That is, instead of using the current URL and appending regular parameters, create a new item selector URL with a criterion with the correct values.

It looks very confusing to me to sometimes use request params, and in other places the criterion values.

Reply:
This is a request to see if that is possible; I'm not asking for that change unconditionally.